src/entry.c

```diff
@@ -976,6 +976,52 @@ vstumpless_set_entry_message( struct stumpless_entry *entry,
   return entry;
 }
 
+const char *
+stumpless_get_entry_procid( const struct stumpless_entry *entry ) {
+  char *procid;
+  
+  VALIDATE_ARG_NOT_NULL( entry );
+
+  procid = alloc_mem( sizeof(char) * 129 );
+
+  if( procid == NULL ) {
+    raise_memory_allocation_failure();
+    return NULL;
+  }
+
+  lock_entry( entry );
+  
+  if( entry->procid_override == false ) {
+    sprintf( procid, "%d" , config_getpid() );
+  }
+  else {
+    sprintf( procid, "%s", (char *) entry->procid );
+  }
+
+  unlock_entry( entry );
+  return procid;
+}
+
+struct stumpless_entry *
+stumpless_set_entry_procid( struct stumpless_entry *entry, const char *procid ) {
+  VALIDATE_ARG_NOT_NULL( entry );
+
+  lock_entry( entry );
+
+  if( procid == NULL ) {
+    sprintf( entry->procid, "%d" , config_getpid() );
```

Comment:
You can remove this line: don't bother modifying the procid field, just signal that it is not valid with the boolean.